src/dask_awkward/layers/layers.py

```diff
@@ -46,25 +46,46 @@ def __init__(
             annotations=None,
         )
 
-    def project_and_mock(self, columns: list[str]) -> AwkwardIOLayer:
+    def __repr__(self):
+        return f"AwkwardIOLayer<{self.output}>"
+
+    def mock(self) -> Any:
 
         # imported here because it this method should be run _only_ on
         # the Client (which is allowed to import awkward)
         from dask_awkward.lib.core import typetracer_from_form
+        import awkward as ak
+        import copy
 
-        new_meta = typetracer_from_form(
-            self._meta.layout.form.select_columns(columns),
-        )
-        return AwkwardIOLayer(
-            name=self.name,
-            columns=self.columns,
-            inputs=[None],
-            io_func=lambda *_, **__: new_meta,
-            label=self.label,
-            produces_tasks=self.produces_tasks,
-            creation_info=self.creation_info,
-            annotations=self.annotations,
-            meta=self._meta,
+        def _label_form(form, start):
+            if form.is_record:
+                for field in form.fields:
+                    _label_form(
+                        form.content(field), f"{start}.{field}"
+                    )
+            elif form.is_numpy:
+                form.form_key = start
+            else:
+                _label_form(form.content, start)
+
+        new_meta = typetracer_from_form(copy.deepcopy(self._meta.layout.form))
+        form = new_meta.layout.form
+        _label_form(form, self.name)
+        new_labelled_meta, report = ak._typetracer.typetracer_with_report(form)
+        nblock = list(self.numblocks.values())[0][0]
+        return (
+            AwkwardIOLayer(
+                name=self.name,
+                columns=self.columns,
+                inputs=[None] * nblock,
```

Comment:
We need all the inputs rather than just the first one now, even though we try only get the first output "partition" while reporting. This is because some actions like aggregations work across all partitions. Can this possibly expand the time it takes to run a graph for optimization purposes?

Reply:
Yea I wonder that as well. We should be able to check this without too much difficulty.